### PR TITLE
TEST: reformat with latest nixfmt

### PR DESCRIPTION
**Pull request by MattSturgeon**
Just to satisfy my curiosity, this is what the repo looks like when running
```
nix run github:nixos/nixfmt -- **/*.nix
```

Note: some of the new formatting uses input-formatting-based heuristics, so if you (for example) wanted to reformat the following, it may require manual intervention.
```nix
# from
mkRenamedOptionModule
  [
    "plugins"
    "hello"
    "foo"
  ]
  [
    "plugins"
    "goodbye"
    "bar"
  ]
# to
mkRenamedOptionModule [ "plugins" "hello" "foo" ] [ "plugins" "goodbye" "bar" ]
# or
mkRenamedOptionModule
  [ "plugins" "hello" "foo" ]
  [ "plugins" "goodbye" "bar" ]
```

We can experiment with additional commits on this branch, manually modifying input formatting and re-running nixfmt to see what it now permits.

### CODE REVIEW

**Comment by khaneliman** [comment]
> ```nix
> mkRenamedOptionModule
>   [ "plugins" "hello" "foo" ]
>   [ "plugins" "goodbye" "bar" ]
> ```

This one looks cleanest to me

**Comment by MattSturgeon** [comment]
Tried playing with some `mkRenameOptionModules`, I was really hoping https://github.com/NixOS/nixfmt/pull/233 would clean these up (as above). So far I'm kinda disappointed:

```nix
(mkRenamedOptionModule [ "foo" "bar" ] [
  "hello"
  "world"
])
```
```nix
(mkRenamedOptionModule [ "foo" "bar" ] (
  foo
  ++ [
    "hello"
    "world"
  ]
))
```
```nix
(mkRenamedOptionModule
  (
    foo
    ++ [
      "bar"
      "baz"
    ]
  )
  [
    "hello"
    "world"
  ]
)
```

The _final_ argument is always expanded to multiple lines, even if the line length would remain short.

Additionally, any concatenation changes the semantics as far as nixfmt is concerned. Concatenation no longer qualifies as a _"list used as a function argument"_; instead it is treated as _"part of a concatenation expression"_. It is therefore always expanded to multiple lines :sob:

This isn't the best place to discuss nix formatting; we should probably raise this as feedback in the nixfmt repo I guess. Although there's already https://github.com/NixOS/nixfmt/issues/206 and https://github.com/NixOS/nixfmt/issues/228, which are related.

_For now_ I'll just cc some of the formatting team:
@infinisil @piegamesde @emilazy

**Comment by emilazy** [comment]
I'm not a member of the formatting team :) though I play one on TV.

**Comment by khaneliman** [comment]
> I'm not a member of the formatting team :) though I play one on TV.

I was going to try blaming you when I saw you tagged but then saw you weren't the one who did the commit :P

**Comment by infinisil** [comment]
With https://github.com/NixOS/nixfmt/pull/257 this is fixed now!

File: plugins/by-name/lazygit/default.nix [hunk 1]
```diff
@@ -47,9 +47,8 @@ lib.nixvim.vim-plugin.mkVimPlugin {
       Config file path is evaluated if this value is `1`.
     '';
 
-    config_file_path = defaultNullOpts.mkNullable (
-      with types; either str (listOf str)
-    ) [ ] "Custom config file path or list of custom config file paths.";
+    config_file_path = defaultNullOpts.mkNullable (with types; either str (listOf str)) [
+    ] "Custom config file path or list of custom config file paths.";
```
Comment (MattSturgeon) [inline plugins/by-name/lazygit/default.nix:51]: TBH, I'm not a huge fan of either of these formats, but breaking within an empty `[ ]` seems particularly strange. Is that intended @infinisil ?

Comment (MattSturgeon) [reply plugins/by-name/lazygit/default.nix:51]: Personally, I'd like to see long multi-arg function calls like this wrapped with one arg per line, e.g.

```nix
config_file_path =
  defaultNullOpts.mkNullable
    (with types; either str (listOf str))
    [ ]
    "Custom config file path or list of custom config file paths.";
```

Comment (piegamesde) [reply plugins/by-name/lazygit/default.nix:51]: Agreed, please open an issue

Comment (MattSturgeon) [reply plugins/by-name/lazygit/default.nix:51]: Sure thing. I've opened https://github.com/NixOS/nixfmt/issues/267 and https://github.com/NixOS/nixfmt/issues/268 for the proposed formatting and weird line-break on empty list, respectively.

**Pull request closed without merging**